### PR TITLE
Produce exportable test DLL

### PR DESCRIPTION
**Pull request by JunielKatarn**
- Produces `React.Windows.Desktop.Test.dll` containing shared test code (i.e. `WebSocketServer` to be consumed by third parties.
- Adds the new DLL to future versions of the NuGet package.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2990)

### CODE REVIEW

File: vnext/Test/React.Windows.Test.vcxproj [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      <Configuration>Debug</Configuration>
+      <Platform>ARM</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|ARM">
+      <Configuration>Release</Configuration>
+      <Platform>ARM</Platform>
```
Comment (NikoAri) [inline vnext/Test/React.Windows.Test.vcxproj]: >ARM [](start = 16, length = 3)

Do we actually run test on arm in this environment?

File: vnext/Desktop.Test.DLL/pch.h [hunk 2]
```diff
@@ -0,0 +1 @@
+#pragma once
```
Comment (NikoAri) [inline vnext/Desktop.Test.DLL/pch.h]: >pragma [](start = 1, length = 6)

Empty pch files are not needed

File: vnext/Test/React.Windows.Test.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Test/React.Windows.Test.vcxproj]: We will.
I will update my `ARM64` branch after this one is merged.

**Pull request merged**